### PR TITLE
RELATED: RAIL-4003 different element definition solution for AttributeFilterButton

### PR DESCRIPTION
**Pull request by plavemar**
- preparation for Cypress E2E tests
- checks for filter compatibility with backend on initialization
- update parent-child filtering with checks for capabilities
- fix invalid attribute elements type resolution

JIRA: RAIL-4003, RAIL-4005

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by plavemar** [comment]
extended check cypress

File: libs/sdk-ui-tests-e2e/scenarios/src/components/constants/attributeFilterButtonConstants.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// (C) 2022 GoodData Corporation
+
+export const TRANSACTION_ATTRIBUTE_ID = "label.salesdetailfact.transactionid";
```
Comment (no23reason) [inline libs/sdk-ui-tests-e2e/scenarios/src/components/constants/attributeFilterButtonConstants.ts:3]: Should we use the exported catalog instead?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6744/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8649/ : FAILURE in 8m 06s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6747/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8651/ : FAILURE in 7m 52s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6760/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8662/ : FAILURE in 7m 47s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6761/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8663/ : FAILURE in 7m 05s

**Comment by plavemar** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6762/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8664/ : SUCCESS in 11m 31s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6763/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/108/ : FAILURE in 7m 37s

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton/hooks/useFetchInitialElements.ts [hunk 2]
```diff
@@ -7,12 +7,13 @@ import stringify from "json-stable-stringify";
...
 import { IAttributeFilterButtonState } from "./useAttributeFilterButtonState";
-import { IAttributeFilter } from "@gooddata/sdk-model";
+import { filterAttributeElements, IAttributeFilter, isAttributeElementsByRef } from "@gooddata/sdk-model";
 
 interface IUseFetchInitialElementsProps {
     context: Omit<AttributeFilterButtonContextProps, "filterObjRef">;
     state: Pick<IAttributeFilterButtonState, "selectedFilterOptions" | "appliedFilterOptions">;
-    callback: (elements: IElementsQueryResult) => void;
+    isElementsByRef: boolean;
+    callback: (elements: IElementsQueryResult, isElementsByRef: boolean) => void;
```
Comment (ivanmjartan) [inline libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton/hooks/useFetchInitialElements.ts:16]: Rename callback to something  more specific.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6765/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8666/ : SUCCESS in 11m 15s

**Review by ivanmjartan — APPROVED** [review]

**Comment by ivanmjartan** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6766/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8667/ : SUCCESS in 11m 38s

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2241

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1992/ : SUCCESS in 9m 58s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2275/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1083/ : SUCCESS in 15m 59s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8669/ : SUCCESS in 11m 33s

**Pull request merged**